File: Array-II/Q3. Find All Numbers Disappeared in an Array/findDisappearedNumbers.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int sz = nums.size();
        vector<int> ans;

        // unordered_map<int,int> mp;
        unordered_set<int> st;

        for(auto num : nums){
            // mp.insert({num, 1});
            st.insert(num);
        }

        for(int i=1; i<=sz; i++){
            // if(mp[i]==0)ans.push_back(i);
            if(!st.count(i))ans.push_back(i);
        }

        return ans;
    }
};

File: Array-II/Q3. Find All Numbers Disappeared in an Array/findDisappearedNumbers.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
    let ans = []
    // let mp = new Map()
    let st = new Set()
    const len = nums.length

    // for(let i=1; i<=len; i++){              //O(n)
    //     if(!nums.includes(i)) ans.push(i)   //O(n)
    // }           //=> TOTAL   =>   O(n^2)


    for (let num of nums) {       //O(n)
        // mp.set(nums[i],1)           //O(1)
        st.add(num)                    //O(1)
    }

    for (let i = 1; i <= len; i++) {      //O(n)
        // if(!mp.has(i))ans.push(i)   //O(1)
        if (!st.has(i)) ans.push(i)      //O(1)
    }

    return ans
};